check_limits.py

```diff
@@ -1,18 +1,29 @@
 
 def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
-
+  return not check_temprature(temperature)
+  return not check_state_of_charge(soc)
+  return not check_charge_rate(charge_rate)
   return True
 
+def print_text(text):
+  print(text)
+
+def check_temprature(temperature):
+  if temperature < 0 or temperature > 45:
+    print_text('Temperature is out of range!')
+    return True
+
+def check_state_of_charge(soc):
+  if soc < 20 or soc > 80:
+    print_text('State of Charge is out of range!')
+    return True
 
+def check_charge_rate(charge_rate):
+  if charge_rate > 0.8:
+    print_text('Charge rate is out of range!')
+    return True
+  return False
+  
 if __name__ == '__main__':
   assert(battery_is_ok(25, 70, 0.7) is True)
```

Comment:
Try to cover boundary conditions, out of range values and in range values, for example test values for temperature could be -1, 0, 25, 45, 50. Same can be applied to other parameters too.